### PR TITLE
Use ClientWidth instead of InnerWidth

### LINKED ISSUE #23390: [Bug]: WithResponsiveMode returning wrong ResponsiveMode values in inspection tools device browser
### Library

Web Components (@fluentui/react)

### System Info

```shell
NA
```

### Are you reporting Accessibility issue?

_No response_

### Reproduction

https://github.com/microsoft/fluentui/blob/ec418075aad560a860b394eeb4ba57b2896c1eed/packages/react/src/utilities/decorators/withResponsiveMode.tsx#L159

## Actual Behavior
On opening the IPhone SE device browser via dev tools the ResponsiveMode returned has a value of `2`

## Expected Behavior
The ResponsiveMode value should be `0`

### Logs

_No response_

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by himanish-star**
## Current Behavior
Whenever `getResponsiveMode` fails to get the `innerWidth` from window, it defaults to a responsive mode equal to `large`

## New Behavior
This change, sets the `_defaultMode` to `responsiveMode` provided as prop, so that in cases of failure scenarios it can default to the component preferred default

## Related Issue(s)

Fixes #23390

### CODE REVIEW

File: packages/react/src/utilities/decorators/withResponsiveMode.tsx [hunk 1]
```diff
@@ -140,6 +142,7 @@ export function getResponsiveMode(currentWindow: Window | undefined): Responsive
     } catch (e) {
       // Return a best effort result in cases where we're in the browser but it throws on getting innerWidth.
       responsiveMode = getInitialResponsiveMode();
+      throw new Error('Failed to get innerWidth on the current window, falling back to default responisve mode');
     }
```
Comment (ThomasMichon) [inline packages/react/src/utilities/decorators/withResponsiveMode.tsx:146]: Rethrowing the error will cause React to de-render all content. I think that's a bit much for simply failing to get the responsive breakpoint.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5f06f19e57ffdc265f10ef26db086bf48dc8f610'>5f06f19e57ffdc265f10ef26db086bf48dc8f610</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/BaseButton-mount_base.html">1060</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/BaseButton-mount.html">1075</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Breadcrumb-mount_base.html">3110</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Breadcrumb-mount.html">3061</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Checkbox-mount_base.html">1738</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Checkbox-mount.html">1733</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/CheckboxBase-mount_base.html">1483</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/CheckboxBase-mount.html">1464</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ChoiceGroup-mount_base.html">5408</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ChoiceGroup-mount.html">5428</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ComboBox-mount_base.html">1108</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ComboBox-mount.html">1108</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/CommandBar-mount_base.html">11944</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/CommandBar-mount.html">11900</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ContextualMenu-mount_base.html">13659</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ContextualMenu-mount.html">13729</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DefaultButton-mount_base.html">1351</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DefaultButton-mount.html">1307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DetailsRow-mount_base.html">4322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DetailsRow-mount.html">4348</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DetailsRowFast-mount_base.html">4362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DetailsRowFast-mount.html">4305</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DetailsRowNoStyles-mount_base.html">4168</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DetailsRowNoStyles-mount.html">4010</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Dialog-mount_base.html">2487</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Dialog-mount.html">2559</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DocumentCardTitle-mount_base.html">194</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/DocumentCardTitle-mount.html">198</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Dropdown-mount_base.html">3814</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Dropdown-mount.html">3795</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/FocusTrapZone-mount_base.html">2028</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/FocusTrapZone-mount.html">2025</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/FocusZone-mount_base.html">1978</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/FocusZone-mount.html">2016</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/IconButton-mount_base.html">2073</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/IconButton-mount.html">2033</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Label-mount_base.html">382</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Label-mount.html">389</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Layer-mount_base.html">3377</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Layer-mount.html">3317</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Link-mount_base.html">535</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Link-mount.html">542</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/MenuButton-mount_base.html">1692</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/MenuButton-mount.html">1700</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/MessageBar-mount_base.html">2442</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/MessageBar-mount.html">2404</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Nav-mount_base.html">3754</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Nav-mount.html">3806</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/OverflowSet-mount_base.html">1232</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/OverflowSet-mount.html">1241</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Panel-mount_base.html">2392</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Panel-mount.html">2386</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Persona-mount_base.html">1159</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Persona-mount.html">1125</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Pivot-mount_base.html">1662</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Pivot-mount.html">1620</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/PrimaryButton-mount_base.html">1496</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/PrimaryButton-mount.html">1513</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Rating-mount_base.html">8952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Rating-mount.html">9071</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SearchBox-mount_base.html">1548</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SearchBox-mount.html">1546</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Shimmer-mount_base.html">2803</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Shimmer-mount.html">2820</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Slider-mount_base.html">2206</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Slider-mount.html">2198</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SpinButton-mount_base.html">5674</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SpinButton-mount.html">5718</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Spinner-mount_base.html">483</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Spinner-mount.html">482</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SplitButton-mount_base.html">3624</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SplitButton-mount.html">3656</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Stack-mount_base.html">600</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Stack-mount.html">610</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/StackWithIntrinsicChildren-mount_base.html">2722</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/StackWithIntrinsicChildren-mount.html">2705</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/StackWithTextChildren-mount_base.html">6224</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/StackWithTextChildren-mount.html">6326</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SwatchColorPicker-mount_base.html">13056</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/SwatchColorPicker-mount.html">12994</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/TagPicker-mount_base.html">3089</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/TagPicker-mount.html">3028</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/TeachingBubble-mount_base.html">110261</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/TeachingBubble-mount.html">112144</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Text-mount_base.html">513</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Text-mount.html">493</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/TextField-mount_base.html">1614</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/TextField-mount.html">1626</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ThemeProvider-mount_base.html">1303</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ThemeProvider-mount.html">1338</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ThemeProvider-virtual-rerender_base.html">706</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ThemeProvider-virtual-rerender.html">683</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2135</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2143</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Toggle-mount_base.html">918</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/Toggle-mount.html">943</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/buttonNative-mount_base.html">138</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23402/perf-test/buttonNative-mount.html">148</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/utilities/decorators/withResponsiveMode.tsx (hunk 1 above)
Comment (himanish-star) [reply packages/react/src/utilities/decorators/withResponsiveMode.tsx:146]: Agree, checking if we can stop this error from occurring at the first place

**Comment by himanish-star** [comment]
using `window.document.documentElement.clientWidth` instead of `window.innerWidth`

**Review by ThomasMichon — APPROVED** [review]

**Comment by himanish-star** [comment]
@JustSlone @micahgodbolt Please review this PR

**Comment by micahgodbolt** [comment]
please run `yarn change` to create a change file for this PR

**Comment by micahgodbolt** [comment]
If anyone comes here with an odd behaving test, do note that testing-library does not support layout, so test will run at smallest breakpoint by default. 

MicrosoftTeams-image (1)

If you need to run your tests at a specific screen width (if the component is responsive) then see this example:

https://github.com/microsoft/fluentui/pull/23402/files#diff-99db1cd5e2e11b0d3e82f7bf56c5416ec447599ba7b78920b6cefd4f43470b1aR34

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**